src/Client.js

```diff
@@ -1,172 +1,240 @@
+ * @param {?Client~observerCallback} options.observer
+ *   Callback that will be called after every completed request.
  */
-export default class Client {
-  /**
-   *
-   * @param {string} options.domain Base URL for the FaunaDB server.
-   * @param {('http'|'https')} options.scheme HTTP scheme to use.
-   * @param {number} options.port Port of the FaunaDB server.
-   * @param {?Object} options.secret
-   *   Auth token for the FaunaDB server.
-   *   Passed straight to [request](https://github.com/request/request#http-authentication).
-   * @param {string} options.secret.user
-   * @param {string} options.secret.pass
-   * @param {?number} options.timeout Read timeout in seconds.
-   * @param {function(res: RequestResult): void} options.observer
-   *   Callback that will be called after every completed request.
-   */
-  constructor(options) {
-    const opts = applyDefaults(options, {
-      domain: 'rest.faunadb.com',
-      scheme: 'https',
-      port: null,
-      secret: null,
-      timeout: 60,
-      observer: null
-    })
-
-    if (opts.port === null)
-      opts.port = opts.scheme === 'https' ? 443 : 80
-
-    this._baseUrl = `${opts.scheme}://${opts.domain}:${opts.port}`
-    this._timeout = Math.floor(opts.timeout * 1000)
-    this._secret = opts.secret
-    this._observer = opts.observer
+function Client(options) {
+  var opts = util.applyDefaults(options, {
+    domain: 'rest.faunadb.com',
+    scheme: 'https',
+    port: null,
+    secret: null,
+    timeout: 60,
+    observer: null
+  });
+
+  if (opts.port === null) {
+    opts.port = opts.scheme === 'https' ? 443 : 80;
   }
 
-  /**
-   * HTTP `GET`.
-   * See the [docs](https://faunadb.com/documentation/rest).
-   * @param {string|Ref} path Path relative the `domain` from the constructor.
-   * @param {Object} query URL parameters.
-   * @return {Promise<Object>} Converted JSON response.
-   */
-  get(path, query=null) {
-    return this._execute('GET', path, null, query)
-  }
+  this._baseUrl = opts.scheme + '://' + opts.domain + ':' + opts.port;
+  this._timeout = Math.floor(opts.timeout * 1000);
+  this._secret = opts.secret;
+  this._observer = opts.observer;
+}
 
-  /**
-   * HTTP `POST`.
-   * See the [docs](https://faunadb.com/documentation/rest).
-   * @param {string|Ref} path Path relative to the `domain` from the constructor.
-   * @param {Object} data Object to be converted to request JSON.
-   * @return {Promise<Object>} Converted JSON response.
-   */
-  post(path, data) {
-    return this._execute('POST', path, data)
-  }
+/**
+ * Executes a query via the FaunaDB Query API.
+ * See the [docs](https://faunadb.com/documentation/queries),
+ * and the query functions in this documentation.
+ * @param expression {Expr}
+ *   The query to execute. Created from query functions such as {@link add}.
+ * @return {external:Promise<Object>} FaunaDB response object.
+ */
+Client.prototype.query = function (expression) {
```

Comment:
I think that is the idea. It keeps the echo principle. 
The rest of the drivers doesn't allow people to create their own expressions neither.